### PR TITLE
Book model

### PR DESCRIPTION
**Pull request by fcavazzoli**
Model Book is working and it is doing pertinents validations

### CODE REVIEW

File: Gemfile [hunk 1]
```diff
@@ -45,12 +45,18 @@ gem 'devise_token_auth'
 # Use Capistrano for deployment
 # gem 'capistrano-rails', group: :development
 
+#Fake generator
+gem 'faker', :git => 'https://github.com/stympy/faker.git', :branch => 'master'
+
+
+
```
Comment (floresv) [inline Gemfile]: Too many empty lines 🤔

**Review by floresv — CHANGES_REQUESTED** [review]
This Pull requests it's mixed with another branch, please clean it or change destination merge.

File: Gemfile [hunk 2]
```diff
@@ -46,12 +46,16 @@ gem 'devise'
...
+#Fake generator
+gem 'faker', :git => 'https://github.com/stympy/faker.git', :branch => 'master'
+
 group :development, :test do
   # Call 'byebug' anywhere in the code to stop execution and get a debugger console
   gem 'byebug', platforms: [:mri, :mingw, :x64_mingw]
   # Adds support for Capybara system testing and selenium driver
   gem 'capybara', '>= 2.15'
   gem 'selenium-webdriver'
+
```
Comment (floresv) [inline Gemfile:55]: It's necessary?

File: app/models/book.rb [hunk 3]
```diff
@@ -0,0 +1,10 @@
+# frozen_string_literal: true
+
+class Book < ApplicationRecord
+  validates :genre, presence: true
```
Comment (floresv) [inline app/models/book.rb]: Maybe [Multiple attributes validations](https://apidock.com/rails/ActiveModel/Validations/ClassMethods/validates#925-multiple-attributes-with-the-same-validations) can improve this code. Remember [DRY](https://en.wikipedia.org/wiki/Don%27t_repeat_yourself).

File: config/database.yml [hunk 4]
```diff
@@ -66,6 +66,8 @@ test:
   pool: 5
   username: wbooks-api
   password: wbooks-api
+  min_messages: WARNING
```
Comment (floresv) [inline config/database.yml]: Why? 🤔

File: db/migrate/20190430212446_create_books.rb [hunk 5]
```diff
@@ -0,0 +1,14 @@
+class CreateBooks < ActiveRecord::Migration[5.1]
+  def change
+    create_table :books do |t|
+      t.string :genre
```
Comment (floresv) [inline db/migrate/20190430212446_create_books.rb]: Remember [Null modifier](https://edgeguides.rubyonrails.org/active_record_migrations.html#column-modifiers)

File: test/fixtures/books.yml [hunk 6]
```diff
@@ -0,0 +1,17 @@
+# Read about fixtures at http://api.rubyonrails.org/classes/ActiveRecord/FixtureSet.html
```
Comment (floresv) [inline test/fixtures/books.yml]: Do you need this file?

File: test/fixtures/users.yml [hunk 7]
```diff
@@ -0,0 +1,19 @@
+# Read about fixtures at http://api.rubyonrails.org/classes/ActiveRecord/FixtureSet.html
```
Comment (floresv) [inline test/fixtures/users.yml]: Do you need this file?

File: test/models/book_test.rb [hunk 8]
```diff
@@ -0,0 +1,7 @@
+require 'test_helper'
```
Comment (floresv) [inline test/models/book_test.rb]: Do you need this file?

File: test/models/user_test.rb [hunk 9]
```diff
@@ -0,0 +1,7 @@
+require 'test_helper'
```
Comment (floresv) [inline test/models/user_test.rb]: Do you need this file?

File: Gemfile [hunk 10]
```diff
@@ -46,12 +46,16 @@ gem 'devise'
 # Use Capistrano for deployment
 # gem 'capistrano-rails', group: :development
 
+#Fake generator
+gem 'faker', :git => 'https://github.com/stympy/faker.git', :branch => 'master'
```
Comment (floresv) [inline Gemfile]: Are you using this gem?

**Review by floresv — CHANGES_REQUESTED** [review]
Please review my comments.

File: config/database.yml (hunk 4 above)
Comment (fcavazzoli) [reply config/database.yml]: this was to prevent my macbook to thow many warnings, but didnt work

File: test/models/book_test.rb (hunk 8 above)
Comment (fcavazzoli) [reply test/models/book_test.rb]: no, when rsepc didnt work as i wanted y tried that out, i can remove it now

File: test/models/user_test.rb (hunk 9 above)
Comment (fcavazzoli) [reply test/models/user_test.rb]: no, when rsepc didnt work as i wanted y tried that out, i can remove it now

File: Gemfile (hunk 10 above)
Comment (fcavazzoli) [reply Gemfile]: yes, i use it to create a fake book

File: config/database.yml (hunk 4 above)
Comment (floresv) [reply config/database.yml]: Then.. remove it, please.

File: test/models/book_test.rb (hunk 8 above)
Comment (floresv) [reply test/models/book_test.rb]: Please, we are gonna work with rspec.

File: app/models/book.rb (hunk 3 above)
Comment (fcavazzoli) [reply app/models/book.rb]: didnt remember that i can do it in just one line

File: test/models/user_test.rb (hunk 9 above)
Comment (floresv) [reply test/models/user_test.rb]: Same that's before.

File: Gemfile (hunk 10 above)
Comment (floresv) [reply Gemfile]: in this PR? 🤔 Where?

File: test/fixtures/books.yml (hunk 6 above)
Comment (fcavazzoli) [reply test/fixtures/books.yml]: deleted

File: test/fixtures/users.yml (hunk 7 above)
Comment (fcavazzoli) [reply test/fixtures/users.yml]: deleted

File: Gemfile (hunk 10 above)
Comment (fcavazzoli) [reply Gemfile]: in null cases

Comment (floresv) [reply Gemfile]: That's right, my bad.

File: db/migrate/20190430212446_create_books.rb [hunk 11]
```diff
@@ -0,0 +1,14 @@
+class CreateBooks < ActiveRecord::Migration[5.1]
+  def change
+    create_table :books do |t|
+      t.string :genre, :null => false, :default => Faker::Book.genre
```
Comment (floresv) [inline db/migrate/20190430212446_create_books.rb]: Why use a faker default? 🤔

File: test/models/book_test.rb [hunk 12]
```diff
@@ -0,0 +1,6 @@
+
```
Comment (floresv) [inline test/models/book_test.rb]: remove it

File: test/models/user_test.rb [hunk 13]
```diff
@@ -0,0 +1,6 @@
+
```
Comment (floresv) [inline test/models/user_test.rb]: remove it

File: db/migrate/20190430212446_create_books.rb (hunk 11 above)
Comment (fcavazzoli) [reply db/migrate/20190430212446_create_books.rb]: just to do not left this empy

**Review by floresv — CHANGES_REQUESTED** [review]
In this card, you shouldn't need faker gem. And remove log development file, please.

File: db/migrate/20190430212446_create_books.rb (hunk 11 above)
Comment (floresv) [reply db/migrate/20190430212446_create_books.rb]: Do you think that this has sense? If I save a book without an author, you add a fake author? 🤔

Comment (fcavazzoli) [reply db/migrate/20190430212446_create_books.rb]: you're right sounds bad idea

File: db/migrate/20190430212446_create_books.rb (hunk 5 above)
Comment (floresv) [reply db/migrate/20190430212446_create_books.rb]: @fedecavazzoli review this, please.

**Pull request merged**